### PR TITLE
Improved documentation regarding distinction of Notebook groups for different notebook servers

### PR DESCRIPTION
**Pull request by AlexandreBrown**
I was trying to figure out why my VSCode code-server custom image was not working (was getting 404 error when navigating to the notebook server).  
Turns out the notebook images group the images are in matter.  
This was not obvious, it was pointed out to me by a slack user , see the thread : https://kubeflow.slack.com/archives/CKBA5D0MU/p1670008626773049  
  
Although the explanations I added is not deep, it would at least prevent users from spending hours trying to figure out why their setup doesn't work and they could get it to work fast.

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Comment by AlexandreBrown** [comment]
@aronchick Hello, I would like someone to take a look at this quick MR whenever possible. Thanks!

**Comment by AlexandreBrown** [comment]
@annajung Hello, do you know who I should ping to get a review for this PR?  
Thanks

File: content/en/docs/components/notebooks/container-images.md [hunk 1]
```diff
@@ -68,8 +68,19 @@ For Kubeflow Notebooks to work with a container image, the image must:
   - the UID of `jovyan` should be `1000`
 - start successfully with an empty PVC mounted at `/home/jovyan`:
   - kubeflow mounts a PVC at `/home/jovyan` to keep state across Pod restarts
-  
+
+### Group Requirements
+- Jupyterlab images should reside in the notebook images group "jupyterlab"  
+![image](https://user-images.githubusercontent.com/26939775/205385486-25750207-be1d-4222-895b-cd69a4fa0e97.png)
```
[the remark this answers is not in the document]
Comment (AlexandreBrown) [reply content/en/docs/components/notebooks/container-images.md]: Thank you for your suggestion I did not know there was a standard in place.  
I have applied your suggestions for all images and moved the images to the repo as well.

File: content/en/docs/components/notebooks/container-images.md [hunk 2]
```diff
@@ -68,8 +68,19 @@ For Kubeflow Notebooks to work with a container image, the image must:
...
+- Visual Studio Code (code-server) images should reside in the notebook images group 1  
+![image](https://user-images.githubusercontent.com/26939775/205385318-2695dc86-98af-4500-b394-cd526e57bf5f.png)
+- RStudio images should reside in the notebook images group 2
+![image](https://user-images.githubusercontent.com/26939775/205385449-29bf5578-54fd-492f-9423-195efce32c62.png)
+If you're using a custom image, you can click on `Custom Image` in the appropriate group to enter it manually when creating a notebook or you can add an entry to the dropdown by adding an option to the appropriate group in the notebook config file : https://github.com/kubeflow/manifests/blob/f40fefafab30834515889d712697e915eeca123b/apps/jupyter/jupyter-web-app/upstream/base/configs/spawner_ui_config.yaml#L18
+#### Reason
```
[the remark this answers is not in the document]
Comment (AlexandreBrown) [reply content/en/docs/components/notebooks/container-images.md]: I agree, this makes more sense and is prettier.  
I have applied your suggestion, please let me know if it is as expected.

File: content/en/docs/components/notebooks/container-images.md [hunk 3]
```diff
@@ -68,8 +68,24 @@ For Kubeflow Notebooks to work with a container image, the image must:
...
 - start successfully with an empty PVC mounted at `/home/jovyan`:
   - kubeflow mounts a PVC at `/home/jovyan` to keep state across Pod restarts
-  
+
+### Group Requirements
+Kubeflow Notebooks images are grouped by server type to ensure networking is configured correctly for Kubeflow usage. Therefore, when using a custom image that is natively supported, it should be manually added to the appropriate group during creation or as an entry to the notebook [configuration file](https://github.com/kubeflow/manifests/blob/f40fefafab30834515889d712697e915eeca123b/apps/jupyter/jupyter-web-app/upstream/base/configs/spawner_ui_config.yaml#L18).
+
+- Jupyterlab images should reside in the notebook images group "jupyterlab"  
+<img src="/docs/images/notebook-images-group-jupyterlab.png" 
+     alt="description"
```
Comment (annajung) [inline content/en/docs/components/notebooks/container-images.md:77]: Can we add alternate text for all the images for accessibility?

**Review by annajung — COMMENTED** [review]
It mostly looks good to me, just added one more comment

deploy preview: https://deploy-preview-3414--competent-brattain-de2d6d.netlify.app/docs/components/notebooks/container-images/#group-requirements

File: content/en/docs/components/notebooks/container-images.md (hunk 3 above)
Comment (AlexandreBrown) [reply content/en/docs/components/notebooks/container-images.md]: Yes great idea.  
I added a small description of what the images represent for all the images.

**Comment by annajung** [comment]
Thanks @AlexandreBrown 
/lgtm

Based on https://github.com/kubeflow/website/blob/master/content/en/docs/components/notebooks/OWNERS, looks like @kimwnasptd @StefanoFioravanzo @thesuperzapper can provide technical review and approvals.

**Comment by thesuperzapper** [comment]
@StefanoFioravanzo @AlexandreBrown @annajung I think this is a good case why we need to merge my PR that improves the comments in the `spawner_ui_config.yaml` file: https://github.com/kubeflow/kubeflow/pull/6736.

Also, @AlexandreBrown feel free to use the language from that PR in your website update, if you want to be really precise about the wording (and what each "group" actually means from a technical perspective).

**Comment by AlexandreBrown** [comment]
@thesuperzapper Thanks for bringing this up.  
Do you think we can merge this PR and incrementally create other PRs to update the doc to add more and more technicalities?  
Since I do not claim to have the expertiese required to author the information you provide in your PR, I would be more comfortable with authoring just this PR as a first step towards better doc then people with more knowledge about the technicalities of each notebook group and their underlying details can create small PRs to improve the doc further.  
  
What do you think?

**Comment by AlexandreBrown** [comment]
@annajung @thesuperzapper @kimwnasptd Hello, I would appreciate if you guys could review this small PR.

I saw someone on slack asking a question where this doc would've help so If we can get it merge it might help others.

Incremental improvements can come in the future.

Cheers

**Comment by AlexandreBrown** [comment]
@DomFleischmann Hello, I am not sure who to ping but I would like to get a review on this PR.   
It has been ready for a while now and I'm sure it could help the community by having this info on the website.  
Do you know who can help with the review process?   
Otherwise I will close this PR as it has been open for months already...  
Thanks

**Pull request closed without merging**